src/ucm/cuda/cudamem.c

```diff
@@ -99,25 +99,13 @@ UCM_DEFINE_REPLACE_DLSYM_PTR_FUNC(cudaMallocPitch, cudaError_t, -1, void**,
 static void ucm_cuda_dispatch_mem_alloc(CUdeviceptr ptr, size_t length,
                                         ucs_memory_type_t mem_type)
 {
-    unsigned sync_atr_value = 1;
-    const char *cu_err_str;
     ucm_event_t event;
-    CUresult ret;
-
-    if ((ptr != 0) && (mem_type == UCS_MEMORY_TYPE_CUDA)) {
-        /* Synchronous operation for GPU direct */
-        ret = cuPointerSetAttribute(&sync_atr_value,
-                                    CU_POINTER_ATTRIBUTE_SYNC_MEMOPS, ptr);
```

Comment:
i think it may still be needed for GPU-direct? or do we have it in UCT?

Reply:
We have this as part of cuda_md mem_query here https://github.com/Akshay-Venkatesh/ucx/blob/topic/disable-whole-alloc/src/uct/cuda/base/cuda_md.c#L136